core/src/storage2/collections/binary_heap/tests.rs

```diff
@@ -343,11 +378,17 @@ fn push_smallest_value_complexity_big_o_1() -> env::Result<()> {
 
 #[test]
 fn pop_complexity_big_o_log_n() -> env::Result<()> {
-    const CONST_READ_WRITES: usize = 2;
+    // 1 wrapper overhead (#508) + wrapper.len + 1 vec overhead (#508) +
+    // 1 vec.len + 1 vec.cell from which to pop
+    const CONST_READS: usize = 5;
+
+    // 1 wrapper.len + 1 vec.len + cell which was modified
+    const CONST_WRITES: usize = 3;
 
     for (n, log_n) in &[(2, 1), (4, 2), (8, 3), (16, 4), (32, 5), (64, 6)] {
-        let expected_reads = log_n * 2 + CONST_READ_WRITES;
-        let expected_writes = log_n + CONST_READ_WRITES;
+        let expected_reads = log_n + CONST_READS;
```

Comment:
So with the overhead obviously very small heaps are more expensive, but perhaps more realistic size heaps are seeing up to a 3rd fewer reads

Items | before | after | improvement
-- | -- | -- | --
8 | 8 | 8 | 0.00%
16 | 10 | 9 | 10.00%
32 | 12 | 10 | 16.67%
64 | 14 | 11 | 21.43%
128 | 16 | 12 | 25.00%
256 | 18 | 13 | 27.78%
512 | 20 | 14 | 30.00%
1024 | 22 | 15 | 31.82%
2048 | 24 | 16 | 33.33%
4096 | 26 | 17 | 34.62%
8192 | 28 | 18 | 35.71%